[feat] Add d3 display for trends in SF and Chicago that are retrieved…

SimonDing87 opened the pull request:
… from mongolab DB

Commit:
[feat] Add d3 display for trends in SF and Chicago that are retrieved from mongolab DB

Commit:
[feat] Add d3 display for trends in SF and Chicago that are retrieved from mongolab DB

polina-munoz commented on server.js:
```diff
@@ -94,30 +106,65 @@ app.get('/test', function(req, res) {
+
+  } // end for loop bracket
 });
 
```
Do you think lines 109-167 could go in their own separate file since they are just calling mongo and not acting as a server? This might be a good candidate for a refactor into the utilities folder under twitterAPI.

polina-munoz commented:
:+1: love the pie charts! :pineapple: :pineapple: 

Just as an fyi you are getting a d3 pie error-- not sure if you want to address that or let it be for now.

<img width="1208" alt="screen shot 2015-12-29 at 12 32 39 am" src="https://cloud.githubusercontent.com/assets/3048412/12030395/bed2bd76-adc3-11e5-8b77-c27784f44320.png">

Other than that, I think the only thing is putting your mongoose code into a separate file would be a good idea before merging to keep our concerns clean (definitely something to work on tomorrow, don't worry about doing it tonight).

We can take a look at this PR tomorrow and then I'm hoping we can deploy what we have by the end of tomorrow. :dancer:

kcvan commented:
hihi! yeah we couldn't figure out how to refresh the charts on click, they keep stacking on top of each other and...doubling the numbers HAHA. pain.

Commit:
[cleanup] Resolve merge conflicts

Commit:
[cleanup] Resolve merge conflicts

SimonDing87 commented:
I might close this PR and submit a new one from Kevin's repo since we will be working on his, and his repo is the same as mine.  Will post updates!

Commit:
[refactor] fix duplication of data in pie chart

Commit:
[refactor] fix duplication of data in pie chart

Commit:
[feat/refactor] Add rest of cities to view, refactor angular code to a factory file

Commit:
[feat/refactor] Add rest of cities to view, refactor angular code to a factory file

Commit:
[refactor] Refactor get city trends to twitterAPI folder

Commit:
[refactor] Refactor get city trends to twitterAPI folder

Commit:
[refactor] Rename routes to be more descriptive

Commit:
[refactor] Rename routes to be more descriptive

SimonDing87 commented:
hey Polina, Kevin and I worked some more on this.  We separated the api call into its own file (cityTrends.js) and took it out of server.js.  Hope all looks good!

polina-munoz commented:
I'll take a look in a bit! :)

polina-munoz commented on server.js:
```diff
@@ -12,6 +13,7 @@ var LocationTrend = require('./mongodb/models/locationTrend.js');
 **************************************************************/
 var countryTrends = require('./utilities/twitterAPI/countryTrends');
+var cityTrends = require('./utilities/twitterAPI/cityTrends');)
 var usRoutes = require('./utilities/middleware/usRoutes');
```
@SimonDing87 @kcvan found some extra semicolons on line 16, not sure if the app will ignore that or crash, but we should probably remove it :)

polina-munoz commented:
Aside from rouge semi colon, looks great! When you mentioned the double data issue, is this what you were referring to? This is what happened when I selected San Francisco:

<img width="847" alt="screen shot 2016-01-02 at 5 23 44 pm" src="https://cloud.githubusercontent.com/assets/3048412/12076906/c49697f4-b175-11e5-9338-eff35f75269e.png">

kcvan commented:
that's really weird! we make it do the div destroyed itself when calling it again. i was working on ours :x

Commit:
[cleanup] Fix syntax error

Commit:
[cleanup] Fix syntax error

Commit:
[refactor] Refactor d3pie so all cities share same code

Commit:
[refactor] Refactor d3pie so all cities share same code

SimonDing87 commented:
I fixed that semi colon error and refactored a bit more.  That's really strange that you're still getting dupe data.  Did you click it multiple times for it to show like that, or was it first click?  I'll have to look into it more.

and just to note we actually don't destroy the div -- we select the div and empty out all of its contents via $('#pieChart').empty()

Commit:
[cleanup] Fix jshint syntax issues

SimonDing87 commented:
Just a quick note for my last commit: it's pretty useful to run "grunt jshint" to catch some missing semicolons and stuff

Sorry this PR is turning out to be a huge beast lol

polina-munoz commented:
I'll download the PR again and check to see what I did to make it look like that, not sure if I did something extra.

polina-munoz commented:
Hey @SimonDing87 can you pull in upstream development changes?

SimonDing87 commented:
oh ya sure.  There are a few conflicts let me resolve them before i push

Commit:
[cleanup] Resolve merge conflicts, fix some jshint syntax errors

Commit:
[cleanup] Resolve merge conflict

SimonDing87 commented:
Okay!  I thinkkkkk that did the trick.  Please let me know if there are any more issues

polina-munoz commented:
@SimonDing87 still merge conflicts :( lmk if i can help

Commit:
[cleanup] Fix merge conflicts

SimonDing87 commented:
Okay finally I think I figured out how to resolve the conflicts.  I was trying to --rebase the entire time and eventually gave up and just did a normal pull lol.  I resolved a few things in bower.json, index.html, and your stackedAreaChartController.js

Hope it all works out now

polina-munoz commented:
sweet! merging now
